### PR TITLE
Student - Automation - Settings, Help, Assignments tests

### PR DESCRIPTION
**Pull request by ndrsszsz**
New E2E test cases for Settings, Help, Assignments

- testRestrictedSubmissionTypesDoesNotAllowStudio - Covers the issue where restricted file type submission settings still allowed Studio
- testHelpPage - Extended with a step for testing custom link
- testPairWithObserverDisabled - Tests if "Pair With Observer" menu item is not available when self registration is not allowed in account settings

### CODE REVIEW

**Comment by inst-danger** [comment]
<details><summary>Student Build QR Code:</summary>

</details>

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>This pull request will not generate a release note.</td>
    </tr>
  </tbody>
</table>


#### Affected Apps: None

[MBL-17343](https://instructure.atlassian.net/browse/MBL-17343)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 91.63% | 91.63% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 8d59f3de5928df6736c04ebf1b37d2f4c570c402
</p>

File: Core/TestsFoundation/DataSeeding/AuthenticationProviders/CreateDSAuthenticationProvidersRequest.swift
Comment (vargaat) [inline Core/TestsFoundation/DataSeeding/AuthenticationProviders/CreateDSAuthenticationProvidersRequest.swift:1]: The file also should be renamed.

**Review by vargaat — CHANGES_REQUESTED** [review]

File: Core/TestsFoundation/DataSeeding/AuthenticationProviders/CreateDSAuthenticationProvidersRequest.swift
Comment (ndrsszsz) [reply Core/TestsFoundation/DataSeeding/AuthenticationProviders/CreateDSAuthenticationProvidersRequest.swift:1]: Thanks, done.

File: Core/TestsFoundation/DataSeeding/AuthenticationProviders/DataSeeder+DSAuthenticationProvider.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+extension DataSeeder {
+    @discardableResult
+    public func setSelfRegistrationForCanvas(selfRegistration: DSSelfRegistration) -> DSAuthenticationProvider {
+        let body = UpdateDSAuthenticationProviderRequest.Body(selfRegistration: selfRegistration)
+        let request = UpdateDSAuthenticationProviderRequest(body: body)
```
Comment (rh12) [inline Core/TestsFoundation/DataSeeding/AuthenticationProviders/DataSeeder+DSAuthenticationProvider.swift:23]: nit: IMHO we could simply do it like this, without sacrificing readability:
```
let request = UpdateDSAuthenticationProviderRequest(body: .init(selfRegistration: selfRegistration))
```
(no need to change it unless you push something else anyway)

**Review by szabinst — APPROVED** [review]

**Review by rh12 — APPROVED** [review]

File: Core/TestsFoundation/DataSeeding/AuthenticationProviders/DataSeeder+DSAuthenticationProvider.swift (hunk 1 above)
Comment (ndrsszsz) [reply Core/TestsFoundation/DataSeeding/AuthenticationProviders/DataSeeder+DSAuthenticationProvider.swift:23]: I like your suggestion but wanted to stay consistent with other DataSeeding solutions so I'd leave it like this for now.

**Pull request merged**